engine/run_ensemble.py

```diff
@@ -73,16 +73,23 @@ def prepare_perturbed_run_script(
     rhs_old = [None if r == "None" else r for r in rhs_old]
 
     for line in in_file:
-        # replace input directory with the ones given in config file
-        for lh, rh_old, rh_new in zip(lhs, rhs_old, rhs_new):
-            out_line = replace_assignment(line, lh, rh_new, rh_old, seed)
-            # replace first match
-            if out_line != line:
-                break
-
-        # rename the experiment name
-        if line == out_line:
-            out_line = replace_string(line, experiment_name, modified_experiment_name)
+        out_line = line
+        # only modify namelist if lhs,rhs_old or rhs_new are not equal None
+        if not all(
+            all(item is None for item in entry) for entry in [lhs, rhs_old, rhs_new]
+        ):
```

Comment:
Maybe a matter of taste, but I would go with something like this:
```if any(entry is not None for entry in [lhs, rhs_old, rhs_new]):```
But I would definelty remove the if statement outside of the loop, it is constant for the whole loop.